lib/IRGen/IRGenSIL.cpp

```diff
@@ -5113,6 +5114,17 @@ void IRGenSILFunction::visitDebugValueInst(DebugValueInst *i) {
                                AddrDbgInstrKind(i->getWasMoved()));
 }
 
+void IRGenSILFunction::visitDebugStepInst(DebugStepInst *i) {
+  // Unfortunately there is no LLVM-equivalent of a debug_step instruction.
+  // Also LLVM doesn't provide a plain NOP instruction.
+  // Therefore we have to solve this with inline assembly.
+  // Strictly speaking, this is not architecture independent. But there are
+  // probably few assembly languages which don't use "nop" for nop instructions.
+  auto *AsmFnTy = llvm::FunctionType::get(IGM.VoidTy, {}, false);
+  auto *InlineAsm = llvm::InlineAsm::get(AsmFnTy, "nop", "", true);
+  Builder.CreateAsmCall(InlineAsm, {});
```

Comment:
A slightly more portable way to do this would be to emit a function call to a `nodebug` function that just returns. But that would increase code size significantly, so then we'd want to make this an always_inline function, and then we'd end up with what is just a slightly more complicated way writing the same code :-)